src/react.tsx

```diff
@@ -6,6 +6,29 @@ import { equals, getDisplayName, keys, mapValues, some } from './utils'
 
 export type Diff<T, U> = Pick<T, Exclude<keyof T, keyof U>>
 
+class StoreSnapshotWrapper<StoreState extends object> implements Store<StoreState> {
```

Comment:
What do you think about just adding `onGet` as a constructor param for [`StoreSnapshot`](https://github.com/bcherny/undux/blob/acd8522fe4676166667006f9fcd28048df4446a6/src/index.ts#L26)? My reasoning is that an extra param is better than a proliferation of things that implement `Store`.

Reply:
I like this idea, onGet and onGetAll

Reply:
I don't think this is possible because there is only one copy of `StoreSnapshot` at any given point.

Reply:
That's a good point @anarelion. We have:

- StoreDefinition - One per Container
- StoreSnapshot - One per StoreDefinition state
- StoreSnapshotWrapper - One per (Component, StoreSnapshot) tuple